### ISSUE TITLE
Level three nodes cant have children

### ISSUE THREAD

**Issue by matiasgarcia** [pqx/react-ui-tree#21]
Consider the case of the following Tree:

Root 
 - Level 1
    - Level 2 Node
    - Another Level 2 Node
 - New node

The component will not let me drag "New node" into any level 2 node. To do this, I have to drag out a level 2 node, drag the "New node" into it and then drag it back into level 1.

I have tried to debug why this is happening but couldn't find any solution. In the github page example, this behaviour happens too.

**Comment by tobyndockerill** [comment]
This is a result of the node having a `leaf: true` property.
If you set this to false (or don't give it the property at all) you will be able to drag nodes under it.

**Comment by matiasgarcia** [comment]
No, this bug was found on the version before the leaf was added.
I updated it and it still happens.

You can replicate this on the demo.

**Comment by matiasgarcia** [comment]
Ok, looks like I found the problem, not sure if it is a bug or desired behavior.

Since my tree is really big, in the initial render I collapsed every single node it has. As I said before, this has been happening before the leaf feature was added.

In the react-ui-tree.js.jsx, the problem is in the drag:

```javascript
if(index.prev) {
                var prevNode = tree.getIndex(index.prev).node;
                if (!prevNode.collapsed && !prevNode.leaf) {
                    newIndex = tree.move(index.id, index.prev, 'append');
                }
            }
        }
```

So, if you have a node that is not a leaf but it is collapsed, the node won't accept any children.

**Comment by wangzuo** [comment]
> The component will not let me drag "New node" into any level 2 node

The level2 node has `leaf: true` ?

**Comment by matiasgarcia** [comment]
Nope, as I said before, all tree nodes from level 1 to the end had collapse on true

**Comment by sergeycooper** [comment]
@matiasgarcia Did you find any solution?

**Comment by matiasgarcia** [comment]
@sergeycooper Since I had a really big tree to render, I collapsed every single node on initialization. The issue was that I also collapsed the leaf nodes on initialization too, and this caused that new nodes couldnt be appended to them (check the code above).

**Comment by sergeycooper** [comment]
@matiasgarcia Alright, I get it. Thanks for info!
Are you still using it or you find any other DnD-Tree-like solution?

**Comment by matiasgarcia** [comment]
Although it has some perfomance issues to be solved when rendering a really big tree, I haven't found anything better. Let me know if you do :)

**Comment by sergeycooper** [comment]
If you haven't found anything better I think I will use it for my purposes too ;)
